joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/dataflow/SemanticTests.scala

```diff
@@ -0,0 +1,126 @@
+package io.joern.javasrc2cpg.querying.dataflow
+
+import io.joern.javasrc2cpg.testfixtures.JavaDataflowFixture
+import io.joern.dataflowengineoss.language._
+import io.joern.dataflowengineoss.queryengine.{EngineContext, EngineConfig}
+
+import scala.jdk.CollectionConverters._
+
+import io.joern.dataflowengineoss.DefaultSemantics
+import io.joern.dataflowengineoss.semanticsloader.FlowSemantic
+
+class SemanticTests
+    extends JavaDataflowFixture(extraFlows =
+      List(
+        FlowSemantic("Test.sanitize:java.lang.String(java.lang.String)", List((0, 0), (1, 1))),
+        FlowSemantic("^ext\\.Library\\.taintReturn:.*", List((0, 0), (1, -1)), regex = true),
+        FlowSemantic("^ext\\.Library\\.taintNone:.*", List((0, 0), (1, 1)), regex = true),
+        FlowSemantic("^ext\\.Library\\.taint1to2:.*", List((1, 2)), regex = true)
+      )
+    ) {
+  behavior of "Dataflow through custom semantics"
+
+  // new Parser().parse("""
```

Comment:
Can we get rid off this commented out code?

Reply:
Yeah sure, I'll add more tests and clean those comments by the way.

Reply:
See: https://github.com/joernio/joern/pull/2587